xapian-core/include/xapian/weight.h

```diff
@@ -409,6 +409,11 @@ class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
     /// The factor to multiply with the weight.
     double factor;
 
+    /// Parameters slope and delta in the Piv+ normalization weighting formula.
+    /// Default values: 0.2 and 1.0 respectively.
+    double param_slope = 0.2;
+    double param_delta = 1.0;
+
     TfIdfWeight * clone() const;
```

Comment:
The constructor just taking a `const std::string&` takes non-pivoted normalizations that don't require `param_slope` and `param_delta` and so does default constructor. Making a way for initialisation of `param_slope` and `param_delta` requires changing the parameter list of this constructor which would break existing tests using this constructor e.g. `TfIdfWeight("ntn")`.

I think we can do the initialisation in overloaded constructor at line [477](https://github.com/xapian/xapian/pull/115/files#diff-8a2d77838da5b7da44ddb21e3d97acb1R477) as:

```c++
TfIdfWeight(const std::string &normalizations, double slope = 0.2, double delta = 1.0);